src/VisualStudio/VisualBasic/Impl/ProjectSystemShim/ConvertedVisualBasicProjectOptions.vb

```diff
@@ -77,20 +77,32 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.ProjectSystemShim
                     kind = OutputKind.WindowsRuntimeMetadata
             End Select
 
+            ' GetSDKPath can return E_NOTIMPL if there is no SDK path at all
+            Dim sdkPath As String = Nothing
+            Dim sdkPathHResult = compilerHost.GetSdkPath(sdkPath)
+
+            If sdkPathHResult = VSConstants.E_NOTIMPL Then
+                sdkPath = Nothing
+            Else
+                Marshal.ThrowExceptionForHR(sdkPathHResult, New IntPtr(-1))
```

Comment:
We previously rejected a PR which had this pattern (#3423).  Are the reasons for rejecting that PR applicable to this change?